genny/auth/auth.go

```diff
@@ -0,0 +1,152 @@
+package auth
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/gobuffalo/buffalo-auth/transformer"
+	"github.com/gobuffalo/buffalo/generators"
+	"github.com/gobuffalo/buffalo/meta"
+	"github.com/gobuffalo/genny"
+	"github.com/gobuffalo/genny/movinglater/plushgen"
+	"github.com/gobuffalo/packr"
+	"github.com/gobuffalo/plush"
+	"github.com/markbates/inflect"
+	"github.com/pkg/errors"
+)
+
+// New actions/auth.go file configured to the specified providers.
+func New(args []string) (*genny.Generator, error) {
+	g := genny.New()
+
+	fields, extraFields := extractFields(args)
+
+	commandParts := append([]string{"db", "generate", "model"}, fields...)
+	g.Command(exec.Command("buffalo", commandParts...))
+
+	if err := g.Box(packr.NewBox("./templates")); err != nil {
+		return g, errors.WithStack(err)
+	}
+
+	ctx := plush.NewContext()
+	ctx.Set("app", meta.New("."))
+
+	g.Transformer(plushgen.Transformer(ctx))
+	g.RunFn(formFieldsFn(extraFields))
+	g.RunFn(addAppActions)
+	g.RunFn(addPasswordFields)
+	g.RunFn(modifyUsersModel)
+
+	return g, nil
+}
+
+func extractFields(args []string) ([]string, []string) {
+	fields := []string{"user", "email", "password_hash"}
+	extraFields := []string{}
+	for _, field := range args {
+
+		fieldName := strings.Split(field, ":")[0]
+		if strings.Contains(strings.Join(fields, "\n"), fieldName) {
+			continue
+		}
+
+		fields = append(fields, field)
+		extraFields = append(extraFields, field)
+	}
+
+	return fields, extraFields
+}
+
+func formFieldsFn(extraFields []string) genny.RunFn {
+	return func(r *genny.Runner) error {
+		fieldInputs := []string{}
+		for _, field := range extraFields {
+			name := inflect.Camelize(field)
+			fieldInputs = append(fieldInputs, fmt.Sprintf(`<%%= f.InputTag("%v", {}) %%>`, name))
+		}
+
+		tr := transformer.NewTransformer("templates/users/new.html")
+		tr.AppendAfter(`<%= f.InputTag("PasswordConfirmation", {type: "password"}) %>`, fieldInputs...)
+
+		return nil
+	}
+}
+
+func addAppActions(r *genny.Runner) error {
+	return generators.AddInsideAppBlock(
```

Comment:
Please use https://godoc.org/github.com/gobuffalo/genny/movinglater/gotools instead of the `buffalo/generators` package. We're trying to get rid of it. :)